src/rviz/InteractiveMarkerViewer.cpp

```diff
@@ -49,6 +49,39 @@ FrameMarkerPtr InteractiveMarkerViewer::addFrame(
   return marker;
 }
 
+
+
+void InteractiveMarkerViewer::visualizeTSR(const aikido::constraint::TSR& _tsr,
+  int nSamples)
+{
+  using dart::dynamics::Frame;
+  using dart::dynamics::SimpleFrame;
+  using aikido::constraint::TSR;
+
+  auto sampler = _tsr.createSampleGenerator();
+  auto state = _tsr.getSE3()->createState();
+
+  for(int i = 0; i < nSamples && sampler->canSample(); ++i)
+  {
+    sampler->sample(state);
```

Comment:
Check the return value of `sample`. In principle it should not return `false` if `canSample()` returned `true`, so an assertion would be fine.